omod-1.8/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_8/TaskActionController1_8Test.java

```diff
@@ -0,0 +1,137 @@
+	        "org.openmrs.scheduler.tasks.TestTask");
+	
+	private String testTaskMockJson = "{\"id\" : 1,\"name\":\"TestTask\",\"description\":\"TestTask Description\","
+	        + " \"taskClass\":\"org.openmrs.scheduler.tasks.TestTask\"}";
+	
+	private MockTaskServiceWrapper mockTaskServiceWrapper = new MockTaskServiceWrapper();
+	
+	@Before
+	public void setUp() throws Exception {
+		mockTaskServiceWrapper.registeredTasks.add(testTask.getName());
+		
+		TaskActionResource1_8 taskActionResource = (TaskActionResource1_8) restService
+		        .getResourceBySupportedClass(TaskAction.class);
+		taskActionResource.setTaskServiceWrapper(mockTaskServiceWrapper);
+		
+		TaskDefinitionResource1_8 taskResource = (TaskDefinitionResource1_8) restService
+		        .getResourceBySupportedClass(TaskDefinition.class);
+		taskResource.setTaskServiceWrapper(mockTaskServiceWrapper);
+	}
+	
+	@Test
+	public void shouldStartTestTask() throws Exception {
+		//sanity check
+		assertThat(mockTaskServiceWrapper.scheduledTasks, not(hasItem(testTask.getName())));
+		deserialize(handle(newPostRequest(getURI(), "{\"action\": \"scheduletask\", \"tasks\":[" + testTaskMockJson + "]}")));
+		assertThat(mockTaskServiceWrapper.scheduledTasks, hasItem(testTask.getName()));
+	}
+	
+	@Test
+	public void shouldStopTestTask() throws Exception {
+		mockTaskServiceWrapper.scheduledTasks.add(testTask.getName());
+		deserialize(handle(newPostRequest(getURI(), "{\"action\": \"shutdowntask\", \"tasks\":[" + testTaskMockJson + "]}")));
+		assertThat(mockTaskServiceWrapper.scheduledTasks, not(hasItem(testTask.getName())));
+	}
+	
+	@Test
+	public void shouldDoNothingIfTestTaskAlreadyStarted() throws Exception {
+		mockTaskServiceWrapper.scheduledTasks.add(testTask.getName());
+		//sanity check
+		assertThat(mockTaskServiceWrapper.scheduledTasks, hasItem(testTask.getName()));
+		assertThat(mockTaskServiceWrapper.registeredTasks, hasItem(testTask.getName()));
+		deserialize(handle(newPostRequest(getURI(), "{\"action\": \"scheduletask\", \"tasks\":[" + testTaskMockJson + "]}")));
+		//check if state preserved
+		assertThat(mockTaskServiceWrapper.scheduledTasks, hasItem(testTask.getName()));
+		assertThat(mockTaskServiceWrapper.registeredTasks, hasItem(testTask.getName()));
+	}
+	
+	@Test
+	public void shouldDoNothingIfTestTaskAlreadyStopped() throws Exception {
+		//sanity check
+		assertThat(mockTaskServiceWrapper.scheduledTasks, not(hasItem(testTask.getName())));
+		assertThat(mockTaskServiceWrapper.registeredTasks, hasItem(testTask.getName()));
+		deserialize(handle(newPostRequest(getURI(), "{\"action\": \"shutdowntask\", \"tasks\":[" + testTaskMockJson + "]}")));
+		//check if state preserved
+		assertThat(mockTaskServiceWrapper.scheduledTasks, not(hasItem(testTask.getName())));
+		assertThat(mockTaskServiceWrapper.registeredTasks, hasItem(testTask.getName()));
+	}
+	
+	@Test
+	public void shouldUnloadTestTask() throws Exception {
```

Comment:
What should unload test task?

Reply:
I used this method to confirm the task deletion. So I named as shouldUnlodTestTask. 
Is it wrong?

Reply:
If the test is about ensuring that the delete action should actually delete the task, then the name could be something like: shouldDeleteTask()

Reply:
renamed :-)